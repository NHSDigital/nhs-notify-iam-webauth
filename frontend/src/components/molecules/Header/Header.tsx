import concatClassNames from '@/src/utils/concat-class-names';
import content from '@/src/content/content';
import { HeaderType } from './header.types';
import styles from './Header.module.scss';
import { AuthLink } from '../AuthLink/AuthLink';

export function NHSNotifyHeader({ dataTestId }: HeaderType) {
  return (
    <header
      className='nhsuk-header'
      role='banner'
      data-testid={dataTestId ?? 'page-header'}
    >
      <div
        className={concatClassNames(
          'nhsuk-header__container',
          styles.header__container
        )}
      >
        <div className='nhsuk-header__logo'>
          <a
            className='nhsuk-header__link nhsuk-header__link--service'
            href={`${content.components.headerComponent.serviceLink}`}
            aria-label='NHS homepage'
          >
            <svg
              className='nhsuk-logo'
              xmlns='http://www.w3.org/2000/svg'
              viewBox='0 0 40 16'
              height='40'
              width='100'
              data-testid='page-header-logo'
            >
              <path
                className='nhsuk-logo__background'
                fill='#005eb8'
                d='M0 0h40v16H0z'
              />
              <path
                className='nhsuk-logo__text'
                fill='#fff'
                d='M3.9 1.5h4.4l2.6 9h.1l1.8-9h3.3l-2.8 13H9l-2.7-9h-.1l-1.8 9H1.1M17.3 1.5h3.6l-1 4.9h4L25 1.5h3.5l-2.7 13h-3.5l1.1-5.6h-4.1l-1.2 5.6h-3.4M37.7 4.4c-.7-.3-1.6-.6-2.9-.6-1.4 0-2.5.2-2.5 1.3 0 1.8 5.1 1.2 5.1 5.1 0 3.6-3.3 4.5-6.4 4.5-1.3 0-2.9-.3-4-.7l.8-2.7c.7.4 2.1.7 3.2.7s2.8-.2 2.8-1.5c0-2.1-5.1-1.3-5.1-5 0-3.4 2.9-4.4 5.8-4.4 1.6 0 3.1.2 4 .6'
              />
            </svg>
            <span className='nhsuk-header__service-name'>
              {content.components.headerComponent.serviceName}
            </span>
          </a>
        </div>
        <div className='nhsuk-header__content' id='content-header'>
          <AuthLink />
        </div>
      </div>
    </header>
  );
}
