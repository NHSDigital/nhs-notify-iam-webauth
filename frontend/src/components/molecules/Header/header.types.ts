export type HeaderType = {
  className?: string;
  dataTestId?: string;
};
