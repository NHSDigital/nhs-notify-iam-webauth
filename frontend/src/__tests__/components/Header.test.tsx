import { render } from '@testing-library/react';
import { NHSNotifyHeader } from '@/src/components/molecules/Header/Header';

jest.mock('@/src/components/molecules/AuthLink/AuthLink', () => ({
  AuthLink: () => <div data-testid='auth-link' />,
}));

describe('Header component', () => {
  it('renders component correctly', async () => {
    const container = render(<NHSNotifyHeader />);

    expect(container.asFragment()).toMatchSnapshot();
  });
});
