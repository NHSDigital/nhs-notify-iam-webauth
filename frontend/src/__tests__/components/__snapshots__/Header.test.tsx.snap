// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders component correctly 1`] = `
<DocumentFragment>
  <header
    class="nhsuk-header"
    data-testid="page-header"
    role="banner"
  >
    <div
      class="nhsuk-header__container header__container"
    >
      <div
        class="nhsuk-header__logo"
      >
        <a
          aria-label="NHS homepage"
          class="nhsuk-header__link nhsuk-header__link--service"
          href="/templates/create-and-submit-templates"
        >
          <svg
            class="nhsuk-logo"
            data-testid="page-header-logo"
            height="40"
            viewBox="0 0 40 16"
            width="100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="nhsuk-logo__background"
              d="M0 0h40v16H0z"
              fill="#005eb8"
            />
            <path
              class="nhsuk-logo__text"
              d="M3.9 1.5h4.4l2.6 9h.1l1.8-9h3.3l-2.8 13H9l-2.7-9h-.1l-1.8 9H1.1M17.3 1.5h3.6l-1 4.9h4L25 1.5h3.5l-2.7 13h-3.5l1.1-5.6h-4.1l-1.2 5.6h-3.4M37.7 4.4c-.7-.3-1.6-.6-2.9-.6-1.4 0-2.5.2-2.5 1.3 0 1.8 5.1 1.2 5.1 5.1 0 3.6-3.3 4.5-6.4 4.5-1.3 0-2.9-.3-4-.7l.8-2.7c.7.4 2.1.7 3.2.7s2.8-.2 2.8-1.5c0-2.1-5.1-1.3-5.1-5 0-3.4 2.9-4.4 5.8-4.4 1.6 0 3.1.2 4 .6"
              fill="#fff"
            />
          </svg>
          <span
            class="nhsuk-header__service-name"
          >
            Notify
          </span>
        </a>
      </div>
      <div
        class="nhsuk-header__content"
        id="content-header"
      >
        <div
          data-testid="auth-link"
        />
      </div>
    </div>
  </header>
</DocumentFragment>
`;
